contents/javascript/Javascript-framework-Redux.md

```diff
@@ -0,0 +1,226 @@
+([And this article](https://www.smashingmagazine.com/2016/06/an-introduction-to-redux/))
+
+Now that we know what Redux is, let us take a look at some of its benefits that were made possible with these concepts of Redux. 
+
+1. **State is made predictable**<br/>
+The features of Redux make state predictable: There is always one source of truth in the store. This store is immutable. Reducers are pure functions, always producing the same result for the same state and action. And with these features, it is possible to implement functionality that is otherwise difficult, such as Undo and Redo. <br/><br/> Because neither the view, nor the network callbacks directly write to state. And changes to state via actions are centralized and happen in strict order, there is no need to worry for race conditions in our applications
+
+2. **Easier debug and inspect**<br/>
+Since actions and state are plain objects, it is possible to serialize them and log them. By logging them, it's easy to understand errors in code. You can also print the store within the code itself (for example using `console.log(store.getState())`{.js}). <br/><br/> The Redux DevTools is an amazing tool as well. Used in the browser, it provides features such as a tree visualization of the application state, as well a time travel ability to replay state changes. 
+
+![Redux Devtool](https://user-images.githubusercontent.com/7957859/48663602-3aac4900-ea9b-11e8-921f-97059cbb599c.png)
+
+
+<center>
+
+_Redux DevTools_ <sup>[image source](https://user-images.githubusercontent.com/7957859/48663602-3aac4900-ea9b-11e8-921f-97059cbb599c.png)</sup>
+</center>
+
+3. **Easier testing**<br/>
+It is easy make tests since the reducers used to change the state are pure functions. When writing code with Redux means writing small, pure and isolated reducer functions, we simply write small and isolated tests. Using Redux makes for testable code.
+
+4. **Server-side rendering**<br/>
+Redux can be used for server-side rendering, by sending the store created on the server side in response to the client's server request. Information from the store then can be used for the initial render of the application. 
+
+## Why not to use Redux?
+
+While Redux has its benefits, you shouldn't jump to Redux for every application you make.
+
+As Dan Abramov, one of the creators of Redux, says:
+
+<box light>
+I would like to amend this: don't use Redux until you have problems with vanilla React.
+</box>
+
+ But the general idea on when to use Redux is when:
+
+1. Reasonable amount of data changing over time
+1. You need a single source of truth
+1. Keeping everything in a top-level parent component is not enough
+
+There are many sites and articles speaking of why you shouldn't or when you shouldn't use Redux. If you are interested, the following resources provide more information and thoughts on when Redux is meant to be used:
+
+- [You Might Not Need Redux](https://medium.com/@dan_abramov/you-might-not-need-redux-be46360cf367)
```

Comment:
Let's keep extra reads to the getting started section

Reply:
`###` seems unnecessary for a small subsection, likewise for the `## Other Frontend...`